### PR TITLE
implemented task and tests

### PR DESCRIPTION
**Pull request by ivadimko**

### CODE REVIEW

File: src/generateDoubleFunction.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @param {function} initialFunction
+ *
+ * @return {function}
+ *
+ * const initialFunction = function(x) {
+ *     return x * x
+ * }
+ * const doubledFunction = generateDoubleFunction(initialFunction);
+ * doubledFunction(5) === 50;
```
Comment (roman-mate) [inline src/generateDoubleFunction.js]: suggestion: add comment // 5 * 5 * 2

Comment (roman-mate) [reply src/generateDoubleFunction.js]: maybe add 1 more example.

File: src/generateDoubleFunction.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const generateDoubleFunction = require('./generateDoubleFunction');
+
+test('First call returns function', () => {
```
Comment (roman-mate) [inline src/generateDoubleFunction.test.js]: suggestion: generateDoubleFunction returns a function.

File: src/generateDoubleFunction.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  expect(generateDoubleFunction((x) => x))
+    .toBeInstanceOf(Function);
+});
+
+test('Second call returns doubled value', () => {
```
Comment (roman-mate) [inline src/generateDoubleFunction.test.js]: suggestion: Calling generateDoubleFunction result returns doubled result of the initial function.

**Comment by ivadimko** [comment]
@roman-mate updated https://github.com/mate-academy/js_generate-double-function/pull/1/commits/1270c0958a847ddf201d54ff18d91ddf6fbd77e7

File: src/generateDoubleFunction.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * doubledFunction(5) === 50; // 5 * 5 * 2 = 50
+ * doubledFunction(4) === 32; // 4 * 4 * 2 = 32
+ *
+ * const initialFunction = function(x) {
+ *     return x + x
+ * }
+ * const doubledFunction = generateDoubleFunction(initialFunction);
+ * doubledFunction(5) === 20; // (5 + 5) * 2 = 20
+ * doubledFunction(4) === 16; // (4 + 4) * 2 = 16
+ *
```
Comment (mgrinko) [inline src/generateDoubleFunction.js]: let's put examples before params

File: src/generateDoubleFunction.test.js (hunk 3 above)
Comment (mgrinko) [reply src/generateDoubleFunction.test.js]: generateDoubleFunction(f)(x) === 2 * f(x)

**Pull request merged**